pubsub/client.go

```diff
@@ -85,37 +75,30 @@ type Client struct {
 	// The current state that the client is in.
 	state     *fsm.Machine
 	stateLock *sync.Mutex
-	// Connection params we're subbed to.
-	conn ConnectionParam
+	// Connection we're subbed to.
+	pool *redis.Pool
 	// The subscription client we're currently using.
 	pubsub *redis.PubSubConn
 	// A list of events that we're subscribed to. If the connection closes,
 	// we'll reestablish it and resubscribe to events.
 	subscribed map[string][]*Listener
-	// Reconnection policy.
-	policy ReconnectPolicy
+	// Reconnection policy
+	policy conn.ReconnectPolicy
 	// Channel of sub/unsub tasks
 	tasks chan task
 }
 
 // Creates and returns a new pubsub client client and subscribes to it.
-func New(conn ConnectionParam) *Client {
-	client := &Client{
+func New(pool *redis.Pool, reconnectPolicy conn.ReconnectPolicy) *Client {
+	return &Client{
 		eventEmitter: newEventEmitter(),
 		state:        blueprint.Machine(),
 		stateLock:    new(sync.Mutex),
-		conn:         conn,
+		pool:         pool,
 		subscribed:   map[string][]*Listener{},
+		policy:       reconnectPolicy,
```

Comment:
[Yes](https://github.com/WatchBeam/redutil/blob/feature/connector/pubsub/client.go#L156). The `pubsub.Client` uses the `*redis.Pool` to generate new connections, but it waits using the reconnect policy to add new connections to that pool.